pandas/core/indexes/period.py

```diff
@@ -161,6 +161,21 @@ def to_timestamp(self, freq=None, how="start") -> DatetimeIndex:
         arr = self._data.to_timestamp(freq, how)
         return DatetimeIndex._simple_new(arr, name=self.name)
 
+    # TODO: use @doc(PeriodArray.hour) once mypy supports
```

Comment:
Is there an issue for this? I think we have facilities in pandas._typing to maintain the signature of functions that are being wrapped already

Reply:
not that im aware of.  cc @simonjayhawkins ?

Reply:
there is a mypy issue so would be OK to ignore however it appears that the doc decorator doesn't work with properties anyway, so this would need to be fixed first.

```
...
  File "C:\Users\simon\pandas\pandas\core\indexes\period.py", line 168, in PeriodIndex
    def hour(self) -> Int64Index:
  File "C:\Users\simon\pandas\pandas\util\_decorators.py", line 385, in decorator
    decorated._docstring_components = (  # type: ignore[attr-defined]
AttributeError: 'property' object has no attribute '_docstring_components'
```

and then could do
```
    # error: Decorated property not supported
    # https://github.com/python/mypy/issues/1362
    @doc(PeriodArray.hour)  # type: ignore[misc]
    @property
    def hour(self) -> Int64Index:
        return Int64Index(self._data.hour, name=self.name)
```

Reply:
Could swap the order and do

```
@property
@doc(PeriodArray.hour.fget)
@hour(self) -> Int64Index:
    [...]
```

Note that PeriodIndex.hour gets overwritten by inherit_names, so the docstring is there even if we dont use `@doc` here.